### ISSUE TITLE
Older Anonymous Comments - Now half-aren't

### ISSUE THREAD

**Issue by EliW** [joindin/joindin-web2#806]
Look here: https://joind.in/event/phptek-2018/i-community-and-so-can-you#comment-e66dc3

A comment that was left under the old 'Make This Anonymous' feature.   It's currently showing the user's headshot, which kinda breaks the whole anonymous piece.

**Comment by dstockto** [comment]
@EliW Do you have any examples of truly anonymous comments (like from back-in-the-day when you could log in and mark a comment as anonymous)? The one mentioned is, as far as I can tell, definitely associated to a user, but that user has no username, but does have an email address. I've been trying to figure out how comments used to be marked anonymous, but I think this one is different.

**Comment by e3betht** [comment]
@dstockto - Is this what you are looking for?
https://joind.in/event/madison-php-conference-2014/make-your-code-behave#comment-f3cf66

**Comment by dstockto** [comment]
@e3betht  - As far as I can see, there's no link to the user who made the anonymous comment on that example, so it seems to be truly anonymous. If there was/is a link, I don't know how to find it. The example that Eli sent seems to have had a user who then didn't have a username, but did have an email that was mapped to a Gravatar. If the email wasn't on Gravatar then I don't know that there's any easy way to find those from the web interface.

**Comment by EliW** [comment]
@e3betht - I can look for one, but not now (php[world] next week) ...

However I would assume that you'll need to look in the database to find this info.   As during "era 2" of Joind.in ... when anonymous cowards were removed, but you could choose to make a comment anonymous ... there obviously needed to be a flag in the database of some sort to mark that comment as such.

(Though the 2.0 refactor may have blown all that away?  Which might explain what you are seeing?)

**Comment by EliW** [comment]
What's interesting I just realized, is that that example I gave was 2018 ... there weren't any anonymous comments in 2018.   So obviously that one, in particular, was just a bug of some sort.

**Comment by e3betht** [comment]
@EliW Anonymous comments were not fully turned off until January of 2019, so there were anonymous comments in 2018.

**Comment by heiglandreas** [comment]
Until beginning of this year the "old" website was still online and it might still have been possible to give anonymous comments via that interface. And we usually didn't remove fields from the database due to the rewrite. We extended the DB but - especially because of the legacy site - we very rarely removed stuff to not break anything...

**Comment by dstockto** [comment]
Unless the production structure is different than dev I’m at a loss to figure out what flag would indicate anonymity. I wonder if those comments were all dropped. I looked at this problem from the code side as well as the database side. The anonymous label got written out if the user record for the comment didn’t have a username, but the rest of that template didn’t care which is why it showed anonymous and had a picture - the email address still existed on the user record that had no username. The fix was merged and it seems to have been deployed as the original link now shows anonymous with the anonymous image.
